components/patient-data/rest/src/main/java/org/phenotips/data/rest/internal/DefaultPatientByLabeledExternalIdentifierResourceImpl.java

```diff
@@ -0,0 +1,437 @@
+     * If there are multiple records that exist, then returns an array of links to the patient records.
+     *
+     * @param patients list of patient internal ID's for records that have the matching {@code label} and {@code id}
+     * @param label the name of the label
+     * @param id the id value for the label
+     * @param jsonInput the incoming json from the request, used to update the created patient record
+     * @param shouldCreate config for whether a patient record should be created
+     * @return a {@link Response} with no content if successful, an error code otherwise
+     */
+    private Response returnIfEmptyOrMultipleExistResponse(List<String> patients, String label, String id,
+        JSONObject jsonInput, boolean shouldCreate)
+    {
+        if (patients.isEmpty()) {
+            if (shouldCreate) {
+                return returnCreatePatientResponse(label, id, jsonInput);
+            } else {
+                this.logger.debug("No patient record with label [{}] and corresponding external ID [{}] exists yet",
+                    label, id);
+                return Response.status(Response.Status.NOT_FOUND).build();
+            }
+        } else {
+            this.logger.debug("Multiple patient records ({}) with label [{}] and corresponding external ID [{}]: {}",
+                patients.size(), label, id, patients);
+            return Response.status(300).entity(this.factory.createAlternatives(patients, this.uriInfo)).build();
+        }
+    }
+
+    /**
+     * Tries to create a patient with provided {@code label} and {@code id} and {@code json data}. Returns a
+     * {@link Response} with no content if successful, an error code otherwise.
+     *
+     * @param label the name of the label
+     * @param id the id value for the label; will be overwritten if a label/id pair is specified in {@code jsonInput}
+     * @param jsonInput the incoming json from the request, used to update the created patient record
+     * @return a {@link Response} with no content if successful, an error code otherwise
+     */
+    private Response returnCreatePatientResponse(final String label, final String id, final JSONObject jsonInput)
+    {
+        this.logger.debug("Creating patient record with label [{}] and corresponding external ID [{}]", label, id);
+        try {
+            final User currentUser = this.users.getCurrentUser();
+            if (!this.access.hasAccess(currentUser, Right.EDIT,
+                this.currentResolver.resolve(Patient.DEFAULT_DATA_SPACE, EntityType.SPACE))) {
+                this.logger.error("Edit access denied to user [{}].", currentUser);
+                return Response.status(Response.Status.FORBIDDEN).build();
+            }
+            final Patient patient = this.repository.create();
+            return returnUpdatePatientResponse(patient, label, id, jsonInput, true, PatientWritePolicy.UPDATE);
```

Comment:
If a new patient is created, a CREATED response should be returned with the canonical location for the new patient included.